include/net/net_context.h

```diff
@@ -128,6 +134,14 @@ typedef void (*net_tcp_accept_cb_t)(struct net_context *new_context,
  *
  * @details The connect callback is called after a connection is being
  * established.
+ * For TCP connections, this callback is called by RX thread so its stack and
+ * execution context is used here. The callback is called after the TCP
+ * connection is established or the connection establishment failed.
+ * Do processing inside this callback as fast as possible in order not to block
```

Comment:
Keep processing in the callback minimal to reduce the time spent blocked while handling packets.